utkdigitalinitiatives/utk-fedora-fedoragsearch-solr-config
# DIT-1299: when the _dt fields come home to roost

**Jira Issue**: [DIT-1299](https://jirautk.atlassian.net/browse/DIT-1299)

## What does this Pull Request do?
After our reindexing work, we removed the `_dt` (date typed field) from our Solr index. This PR selectively adds `_dt` fields back.

## What's new?
This is a significant refactoring of several stylesheets.
1. foxmlToSolr:
  * comment/drop unused stylesheet `xsl:include` instructions
  * comment/drop `xslt-date-template` and `xslt-edtf-date-template` stylesheets
 2. slurp_all_MODS_to_solr:
  * comment/drop unused stylesheet `xsl:include` instruction
  * add a new mode, `utk_MODS_dates`
  * refactor `_ms` date fields
  * add new templates for processing `_dt` fields
  * move the functionality of `xslt-date-template` and `xslt-edtf-date-template` stylesheets into named templates in `slurp_all_MODS...`
  * modify functionality of `get_ISO8601_edtf_date` named template to accommodate our local EDTF practices
3. Add a mess of test files under `test_records`.
4. Add the `xslt-edtf-date-template` stylesheet to the repository as a artifactual kind of thing.

## How should this be tested?

1. Add objects to islandora vagrant, using the provided test data or your own favorite MODS records. I used the Basic Image Collection for testing purposes and the following steps will reflect that (see attached image as a dummy file for each record
![Sandworms](https://user-images.githubusercontent.com/2052902/78093464-4488d080-73a0-11ea-98a3-58550da2d83b.jpg)
).
2. Look at the Solr documents (please note that this query only provides results from the basic image collection *and* only displays the PID and date fields):  [http://localhost:8080/solr/collection1/select?q=*%3A*&fq=RELS_EXT_isMemberOfCollection_uri_s%3A%22info%3Afedora%2Fislandora%3Asp_basic_image_collection%22&sort=PID+asc&fl=PID%2C+*date*&wt=json&indent=true](http://localhost:8080/solr/collection1/select?q=*%3A*&fq=RELS_EXT_isMemberOfCollection_uri_s%3A%22info%3Afedora%2Fislandora%3Asp_basic_image_collection%22&sort=PID+asc&fl=PID%2C+*date*&wt=json&indent=true)
3. Rename or remove the original `foxmlToSolr.xslt` and `islandora_transforms` directory at: /var/lib/tomcat7/webapps/fedoragsearch/WEB-INF/classes/fgsconfigFinal/index/FgsIndex/
4. Recursively copy (`cp -r`) this feature branch's `foxmlToSolr.xslt` and `islandora_transforms` directory to: /var/lib/tomcat7/webapps/fedoragsearch/WEB-INF/classes/fgsconfigFinal/index/FgsIndex/
5. In the `foxmlToSolr.xslt`, modify the `xsl:include` instructions' `@href` URIs with the following: `sudo sed -i -e 's|/vhosts/fedora/tomcat/|/var/lib/tomcat7/|g' foxmlToSolr.xslt`
6. Restart tomcat: `sudo service tomcat7 restart`
7. Update gsearch for your object with curl or python or GUI (I've attached the `pid-list.txt` file to the PR; add to your `$VAGRANT` directory):

``` curl
while read line; do curl -X GET -u fedoraAdmin:fedoraAdmin "http://localhost:8080/fedoragsearch/rest?operation=updateIndex&action=fromPid&value=$line"; done < /vagrant/pid-list.txt
```

```python

import requests

my_pid = 'test:1'
requests.post(f'http://localhost:8080/fedorafedoragsearch/rest?operation=updateIndex&action=fromPid&value={my_pid}', auth=('fedoraAdmin', 'fedoraAdmin'))

```
8. Rerun the Solr query from 2 above. Review the changes.

## Additional Notes:
This is a big ugly PR. Please let me know if you have questions.

## Interested parties
@markpbaggett @mlhale7 

[pid-list.txt](https://github.com/utkdigitalinitiatives/utk-fedora-fedoragsearch-solr-config/files/4412508/pid-list.txt)
